feat: enhance UI components and add data table

sauravsid opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Theme switcher (Light/Dark/System), top navigation, and a richer Recent Liquidations table with sorting, filtering, pagination, and global search.
  - CSV export via a dialog with email input and live download.

- **UI**
  - New reusable UI primitives (buttons, badges, inputs, dialogs, popovers, selects, tables, checkbox), network icons, and updated copy-to-clipboard button.
  - Slightly tighter border radius and adjusted background visuals; streamlined homepage layout.

- **Refactor**
  - Centralized types, network helpers, formatting utilities, and a hook for live liquidation data.

- **Chores**
  - Added theming, table, and Radix UI dependencies.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

moose-code commented:
<img width="450" height="206" alt="Screenshot 2025-08-22 at 11 38 23" src="https://github.com/user-attachments/assets/efe982ca-04a1-40c5-b3ef-6e0cc377be73" />
Euelr does not appear in the filters for protocols

moose-code commented:
<img width="186" height="190" alt="Screenshot 2025-08-22 at 11 38 42" src="https://github.com/user-attachments/assets/30974800-853d-4b21-b5f7-0d9de3bb21bf" />
Missing chains too in the filters

moose-code commented:
<img width="275" height="93" alt="Screenshot 2025-08-22 at 11 39 27" src="https://github.com/user-attachments/assets/d7676ab0-6165-41a9-9a9e-6c6d9612ec56" />
Suggestion / default text is cut off here so final bit is not actually helpful. Need to trim or change this.

moose-code reviewed (changes requested):
Really nice stuff! Looks great! I left some screenshots and small comments on some things to change before merging :)

sauravsid commented:
Sorry for the late reply, been swamped with work lately 😅

Thanks for the feedback! addressing your points below: 

- Table width/horizontal scroll issue-
The table is already adjusted to use column element width, but I'll see what else I can do. The horizontal scroll is normal for data tables, especially on smaller screens.

- Missing protocols & chains in filters-
These filters are currently client-side only. I haven't implemented server-side filtering or modified the data fetching logic yet. So you'll only see the chains/protocols that are in the currently loaded data.

- Truncated filter text
I'll shorten the placeholder text
